src/components/diagnosisForm/DiagnosisForm.js

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Alert, FlatList } from "react-native";
+import DiagnosisQuestionItem from "./QuestionItem";
+import TextButton from "../common/TextButton";
+
+const DiagnosisForm = ({ questions, setAnswerByIndex, onSubmit, answers }) => {
+  const submitWithValidation = () => {
+    const checkedAnswers = answers.filter((answer) => answer !== undefined);
+
+    const allCheckboxChecked = checkedAnswers.length === answers.length;
+    if (allCheckboxChecked === true) {
+      onSubmit();
+    } else {
+      // TODO: Set all empty questions labels color to red and add some error message near them
+      Alert.alert("Błąd", "Wykryto brakujące odpowiedzi", [
+        {
+          text: "Popraw",
+          style: "cancel",
+          onPress: () => {},
+        },
+      ]);
+    }
+  };
+
+  return (
+    <FlatList
+      data={questions}
+      keyExtractor={(question) => question.content}
+      renderItem={({ item, index }) => (
+        <DiagnosisQuestionItem
+          question={item}
+          setAnswer={(answer) => {
+            setAnswerByIndex(index, answer);
+          }}
+        />
+      )}
+      ListFooterComponentStyle={{ marginTop: 20 }}
+      ListFooterComponent={
+        <TextButton onPress={submitWithValidation} text="Sprawdź odpowiedzi" />
```

Comment:
Tekst "Sprawdź odpowiedzi" nie bardzo tu pasuje, z czym mielibyśmy to sprawdzać? Może lepszy byłby button ze strzałką naprzód?

Reply:
Właśnie na początku miałem tę strzałkę :D Tylko, że stwierdziłem, że w pierwszym formularzu sprawdzamy czy przejść do pytań szczegółowych - stąd taki napis (chciałem jakąś ikonkę dać, ale nie miałem pomysłu - lepszy tekst też mi nie wpadł do głowy jak masz jakiś pomysł to chętnie to zmienię. 
Druga kwestia to to, że zapomniałem to zmienić w kolejnym formularzu - pojawił się ten problem, o którym pisałem w liście modułów, do końca nie jest reużywalny ten komponent przez to, że ma taki guzik. 
Wprowadzę do niego propsa, w którym będzie można przekazać footer i na ekranie z pytaniami szczegółowymi dam strzałkę